core/controllers/admin.py

```diff
@@ -390,7 +390,12 @@ class DataExtractionQueryHandler(base.BaseHandler):
     @acl_decorators.can_access_admin_page
     def get(self):
         exp_id = self.request.get('exp_id')
-        exp_version = int(self.request.get('exp_version'))
+        exp_version = self.request.get('exp_version')
```

Comment:
exp_version can be None so first we are checking if its not None and if so then we are converting it to int. The strange thing though is that if we pass None as params in self.get_json, it gets passed as a str(unicode to be precise) and not as None type. This is why I explicitly converted it into None

Reply:
We can't execute this block of code is exp version is not None(line 405-407):
```
if exploration is None:
            raise self.InvalidInputException(
                'No exploration with ID \'%s\' exists.' % exp_id)
```
You see the strict flag applies only if version is None:
https://github.com/oppia/oppia/blob/0cc83ad67c5ad5f753224e1f6d345e3443e20472/core/storage/base_model/gae_models.py#L780

Reply:
That's a good catch. Several changes might be needed here:

- First, let's keep the old int casting, but put it in a try/except and raise InvalidInputError (with a clear message to the client) if it fails.
- After that, the only problem that can happen is that the int is out of range (in which case you might have no model to retrieve for the exploration). In this case, we might need to modify get_exploration_by_id to handle these cases correctly. So, what currently happens when you call get_exploration_by_id with an invalid exp_version (i.e. a numeric version that's out of range)?

Reply:
> So, what currently happens when you call get_exploration_by_id with an invalid exp_version (i.e. a numeric version that's out of range)?

We will get an error "Entity for class ExplorationModel with id invalid_exp_id not found". This is from here most probably:
https://github.com/oppia/oppia/blob/e69b57106e42f5b7f80e8556b3bd62713f4fadf2/core/storage/base_model/gae_models.py#L101

This does not depend on strict. Strict only works if version is None. If we are keeping the int construct in a try block, then the block of code I mentioned earlier(the one that raises 'No exploration with ID \'%s\' exists.' % exp_id') would no longer be valid, right? We can then also put get_exploration_by_id in a try block and if that fails, we can raise InvalidInputError("The exploration with version xx is not found')?

Reply:
Yep, I think you can replace it with a try/except check for the "Entity ... not found" error. I think that should cover all the cases.

Reply:
done